97_extra_materials.Rmd

```diff
@@ -290,3 +290,324 @@ e <- attr(reducedDim(tse_Genus, "PCoA_BC"), "eig");
+
+```{r}
+# Loading example data
+# Using GlobalPatterns dataset
+
+data(package = "phyloseq", "GlobalPatterns") # Phyloseq object
+PHY <- GlobalPatterns # Rename
+PHY # Check the phyloseq object
+
+data(package = "mia", "GlobalPatterns") # TreeSE object
+TSE <- GlobalPatterns # Rename
+TSE # Check the TSE object
+```
+
+### Accessing different types of data in phyloseq versus TreeSE
+
+**Accessing the table of taxonomic names: tax_table = rowData**
+
+Phyloseq and TreeSE objects' taxonomy tables can be accessed with tax_table and rowData commands.
+
+```{r}
+PHY_taxtable <-
+  tax_table(PHY)  %>% # Access the phyloseq taxonomic name table
+  data.frame # Make into a data frame
+
+TSE_taxtable <- rowData(TSE) %>% # Same for TSE
+  data.frame
+
+```
+
+**Accessing sample data: sample_data = colData**
+
+Sample data can be accessed with sample_data and colData commands.
+
+```{r}
+PHY_sampledata <-
+  sample_data(PHY) %>% data.frame
+
+TSE_sampledata <-
+  colData(TSE) %>% data.frame
+
+```
+
+**Accessing operational taxonomic unit (OTU) abundance objects: otu_table = assay**
+
+OTU tables can be accessed with otu_table and assay commands. The assay can also hold other types of information like taxa abundances from shotgun metagenomic annotation, or functional gene abundances.
+```{r}
+PHY_otutable <-
+  otu_table(PHY) %>% data.frame
+
+TSE_otutable <-
+  assay(TSE) %>% data.frame
+
+```
+
+### Building phyloseq objects vs TreeSE objects: phyloseq = TreeSummarizedExperiment
+After learning how to access various data types from TreeSE, let's see how creating TreeSE objects compares to creating Phyloseq objects. We will use the vanilla data frames we created from the Phyloseq object to demonstrate making both types od data objects.
+
+
+```{r}
+OTU_PHY <- otu_table(PHY_otutable %>% as.matrix, taxa_are_rows = TRUE) # Make OTU table
+TAX_PHY <- tax_table(PHY_taxtable %>% as.matrix) # Make TAX table
+SAMPLE_PHY <- sample_data(PHY_sampledata) # Make sample data table
+
+PHY <- phyloseq(OTU_PHY, TAX_PHY, SAMPLE_PHY) # Combine into phyloseq object
+PHY # Inspect
+
+```
+
+Let's start by checking our otu table, and see if it is counts or already normalized. We will use the same data frame extracted from the PHY object as before.
+
+```{r}
+# Check if we have counts or normalized data
+
+PHY_otutable %>% head
+
+```
+
+We have counts!
+
+Since TreeSEs can hold many different versions of the OTU table, most commonly either relative abundances or counts, we will need to give our assay a name and list the different types of assays we have. In this example we only have one item 'counts' in the list.
+
+Let's convert the data frame to a matrix and make the list of assays.
+
+```{r}
+# Create TreeSE
+counts <- as.matrix(PHY_otutable) # Convert to a matrix
+assays <- SimpleList(counts = counts)
+
+TSE <- TreeSummarizedExperiment(assays = assays, 
+                                colData = PHY_sampledata,
+                                rowData = PHY_taxtable
```

Comment:
slightly confusing that PHY is somehow needed here. See suggestion above to separate data elements before PHY or TSE construction.